frontend/src/metabase/setup/components/LicenseTokenStep/LicenseTokenStep.tsx

```diff
@@ -14,20 +14,22 @@ import type { NumberedStepProps } from "../types";
 import { LicenseTokenForm } from "./LicenseTokenForm";
 
 export const LicenseTokenStep = ({ stepLabel }: NumberedStepProps) => {
-  const { isStepActive, isStepCompleted, handleStepSelect, isSetupCompleted } =
-    useStep("license_token");
+  const { isStepActive, isStepCompleted } = useStep("license_token");
 
   const storeToken = useSelector(state => state.setup.licenseToken);
 
   const dispatch = useDispatch();
 
-  const handleValidSubmit = (token: string | null) => {
-    dispatch(
-      addUndo({
-        message: t`Your license is activated`,
-      }),
-    );
-    dispatch(submitLicenseToken(token));
+  const handleSubmit = async (token: string | null) => {
+    try {
+      await dispatch(submitLicenseToken(token)).unwrap();
+      dispatch(addUndo({ message: t`Your license is activated` }));
+    } catch (err) {
+      console.error(err);
+      throw new Error(
+        t`This token doesn’t seem to be valid. Double-check it, then contact support if you think it should be working.`,
```

Comment:
There will be follow up work to make this submission provide better error messaging based on the work done by: https://github.com/metabase/metabase/pull/39067/